components/eventing-controller/pkg/backend/nats/jetstream/jetstream_integration_test.go

```diff
@@ -531,15 +547,19 @@ func TestJetStreamSubAfterSync_FilterRemoved(t *testing.T) {
 	// Check if total existing NATS subscriptions are correct
 	require.Len(t, jsBackend.subscriptions, 1)
 	// Verify that the nats subscriptions for first subject was not re-created
-	jsSubject := jsBackend.GetJetStreamSubject(firstSubject)
-	jsSubKey := NewSubscriptionSubjectIdentifier(sub, jsSubject)
-	jsSub := jsBackend.subscriptions[jsSubKey]
-	require.NotNil(t, jsSub)
-	require.True(t, jsSub.IsValid())
+	firstJsSubject := jsBackend.GetJetStreamSubject(firstSubject)
+	firstJsSubKey := NewSubscriptionSubjectIdentifier(sub, firstJsSubject)
+	firstJsSub := jsBackend.subscriptions[firstJsSubKey]
+	require.NotNil(t, firstJsSub)
+	require.True(t, firstJsSub.IsValid())
+	// make sure old filter doesn't have any JetStream consumer
+	secondCon, err := secondJsSub.ConsumerInfo()
+	require.Nil(t, secondCon)
+	require.ErrorIs(t, err, nats.ErrConsumerNotFound)
```

Comment:
I ran the tests `make test-local` with and without the bug fix, and in both cases the tests were passing. Can we make the test fail when the bug fix is not there?